server/src/main/java/org/opensearch/cluster/decommission/DecommissionAttribute.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.decommission;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public final class DecommissionAttribute implements Writeable {
```

Comment:
DecommissionAttribute class seems to be generic class with just key/value pair, nothing specific to Decommissioning. why do we need a class for for it? Also could there be multiple decommissioning attributes?

Reply:
At the moment, we are not supporting multiple attribute decommissioning. This class helps the code to be clean and easy to understand and will also be helpful when we incorporate multiple attribute values  decommissioning, only few nodes with a given attribute decommissioning, etc. 

Also replied here - https://github.com/opensearch-project/OpenSearch/pull/4084#discussion_r957076985